File: Node.JS/skebbyAPI.js
// Skebby Node.js Gateway Class
var qs = require('querystring');
var https = require('https');

var method = skebbyAPI.prototype;

function skebbyAPI(user,pass) {
	this._user = user;
	this._pass = pass;
	this._path = '/api/send/smseasy/advanced/http.php';
	this._host = 'gateway.skebby.it';
}

// the core function
method.do_post_request = function(input,callback,callback_error) {
	
	var res_done = false,
	data = qs.stringify(input),
	client = https.request({
        port:	 443,
        path:	 this._path,
        host:	 this._host,
		method : "POST", 
        headers: { 
            "Content-Type": 	"application/x-www-form-urlencoded",
            "Content-Length": 	data.length,
            "Content-Encoding": "utf8",
			"User-Agent":		"Skebby Node.js Gateway"
        }
    }, function(res){
        var res_data = "";
        res.on("data", function(data) { res_data+=data; });
        res.on("end",function(){
            if (!res_done){
				var res_parsed = qs.parse(res_data);
				if(res_parsed.status == "success") {
					callback({data:res_parsed});
					return;
				} else {
					callback_error({data: res_parsed});
					return;
				}
            	res_done = true;
            }
        });
    }); 
 
    client.end(data);
    client.on('error', function(e) {
        if (!res_done){
            res_done = true;
            callback_error(e);
			return;
        }
    });
};

// send SMS function
method.sendSMS = function(input,callback,callback_error) {
	var data = {
		username:		 this._user,
		password:		 this._pass,
		"recipients[]":	 input.recipients || [],
		text:			 input.text,
		method:			 input.method || "send_sms_classic"
	};
	
	if(input.sender_number) { data.sender_number = input.sender_number; }
	if(input.sender_string) { data.sender_string = input.sender_string; }
	if(input.delivery_start) { data.delivery_start = input.delivery_start; }
	if(input.encoding_scheme) { data.encoding_scheme = input.encoding_scheme; }
	if(input.validity_period) { data.validity_period = input.validity_period; }
	if(input.user_reference) { data.user_reference = input.user_reference; }
	
	this.do_post_request(
		data,
		function(response) {
			callback(res);
			return;
		},
		function(error) {
			callback_error(err);
			return;
		}
	);
};

// get Credit function
method.getCredit = function(charset,callback,callback_error) {
	var data = {
		method:		 "get_credit",
		username:	 this._user,
		password:	 this._pass,
		charset:	 charset || "UTF-8"
	};
	
	this.do_post_request(
		data,
		function(response) {
			callback(res);
			return;
		},
		function(error) {
			callback_error(err);
			return;
		}
	);
};

// add Alias function
method.addAlias = function(input,callback,callback_error) {
	var data = {
		method:			 "add_alias",
		username:		 this._user,
		password:		 this._pass,
		charset:		 charset || "UTF-8",
		alias:			 input.alias,
		business_name:	 input.business_name,
		nation:			 input.nation,
		vat_number:		 input.vat_number,
		taxpayer_number: input.taxpayer_number,
		street:			 input.street,
		city:			 input.city,
		postcode:		 input.postcode,
		contact:		 input.contact
	};
	
	this.do_post_request(
		data,
		function(response) {
			callback(res);
			return;
		},
		function(error) {
			callback_error(err);
			return;
		}
	);
}

module.exports = skebbyAPI;

File: VB.NET/skebbyAPI.vb
Imports System
Imports System.IO
Imports System.Web
Imports System.Net
Imports System.Text
Imports System.Collections
Imports System.Collections.Generic

Namespace skebbyGW

    Public Class skebbyAPI
        Protected Username As String
        Protected Password As String
        Protected Url As String

        Sub New(ByVal user As String, ByVal pass As String)
            MyClass.Username = WebUtility.UrlEncode(user)
            MyClass.Password = WebUtility.UrlEncode(pass)
            MyClass.Url = "http://api.skebby.it/api/send/smseasy/advanced/http.php"
        End Sub

        Protected Function doPostRequest(ByVal data As Dictionary(Of String, Object))
            data.Add("username", MyClass.Username)
            data.Add("password", MyClass.Password)
            If Not data.ContainsKey("charset") Then
                data.Add("charset", "UTF-8")
            End If

            Dim request As WebRequest = WebRequest.Create(MyClass.Url)
            request.Credentials = CredentialCache.DefaultCredentials
            CType(request, HttpWebRequest).UserAgent = "Skebby VB.NET Gateway"
            request.Method = "POST"

            Dim parameters As String = ""

            For Each item As KeyValuePair(Of String, Object) In data
                If item.Key = "recipients" Then
                    If TypeOf item.Value Is List(Of Dictionary(Of String, String)) Then
                        For i As Integer = 0 To (item.Value.Count - 1)
                            For Each rec As KeyValuePair(Of String, String) In item.Value(i)
                                parameters = parameters & "&recipients[" & i & "][" & WebUtility.UrlEncode(rec.Key) & "]=" & WebUtility.UrlEncode(rec.Value)
                            Next
                        Next
                    Else
                        For Each number As String In item.Value
                            parameters = parameters & "&recipients[]=" & WebUtility.UrlEncode(number)
                        Next
                    End If
                Else
                    parameters = parameters & "&" & WebUtility.UrlEncode(item.Key) & "=" & WebUtility.UrlEncode(item.Value)
                End If
            Next
            Console.WriteLine(parameters)
            Dim byteArray As Byte() = Encoding.UTF8.GetBytes(parameters)

            request.ContentType = "application/x-www-form-urlencoded"
            request.ContentLength = byteArray.Length

            Dim dataStream As Stream = request.GetRequestStream()

            dataStream.Write(byteArray, 0, byteArray.Length)
            dataStream.Close()

            Dim response As WebResponse = request.GetResponse()

            dataStream = response.GetResponseStream()

            Dim reader As New StreamReader(dataStream)
            Dim responseFromServer As String = reader.ReadToEnd()

            reader.Close()
            dataStream.Close()
            response.Close()

            Dim result As Dictionary(Of String, String) = New Dictionary(Of String, String)

            If responseFromServer.Length <> 0 Then
                Dim values = responseFromServer.Split(New Char() {"&"c})
                For Each value In values
                    Dim temp = value.Split(New Char() {"="c})
                    result.Add(WebUtility.UrlDecode(temp(0)), WebUtility.UrlDecode(temp(1)))
                Next
            Else
                result.Add("result", "fail")
            End If

            Return result

        End Function

        Public Function sendSMS(
            ByVal input As Dictionary(Of String, Object)
        )
            If Not input.ContainsKey("method") Then
                input.Add("method", "send_sms_classic")
            End If
            Return doPostRequest(input)
        End Function

        Public Function getCredit(
            Optional ByVal charset As String = ""
        )
            Dim input As New Dictionary(Of String, Object)
            input.Add("method", "get_credit")
            If (charset <> "") Then
                input.Add("charset", charset)
            End If

            Return doPostRequest(input)
        End Function

        Public Function addAlias(
            ByVal input As Dictionary(Of String, Object)
        )
            input.Add("method", "add_alias")

            Return doPostRequest(input)
        End Function

        Public Sub printResult(ByVal result As Dictionary(Of String, String))
            For Each item As KeyValuePair(Of String, String) In result
                Console.WriteLine(item.Key & ": " & item.Value)
            Next
        End Sub
    End Class
End Namespace